pkgs/by-name/gh/ghdl/package.nix

```
@@ -47,14 +61,44 @@ stdenv.mkDerivation (finalAttrs: {
     "--enable-synth"
   ] ++ lib.optionals (backend == "llvm") [
     "--with-llvm-config=${llvm.dev}/bin/llvm-config"
+  ] ++ lib.optionals (backend == "gcc") [
+    "--with-gcc=gcc-${gcc-unwrapped.version}"
+  ];
+
+  buildPhase = lib.optionalString (backend == "gcc") ''
+    make copy-sources
+    mkdir gcc-objs
+    cd gcc-objs
+    ../gcc-${gcc-unwrapped.version}/configure \
+      --with-native-system-header-dir=/include \
+      --with-build-sysroot=${lib.getDev stdenv.cc.libc} \
```

It's probably better to accept `glibc` as an argument so users can override it if they want to.
```suggestion
      --with-build-sysroot=${lib.getDev glibc} \
```